csharp/src/Ice/Internal/Instance.cs

```diff
@@ -908,6 +893,41 @@ internal void initialize(Ice.Communicator communicator, Ice.InitializationData i
 
             _retryQueue = new RetryQueue(this);
 
+            string[] retryValues = _initData.properties.getIcePropertyAsList("Ice.RetryIntervals");
+            if (retryValues.Length == 0)
+            {
+                retryIntervals = [0];
+            }
+            else
+            {
+                retryIntervals = new int[retryValues.Length];
+
+                for (int i = 0; i < retryValues.Length; i++)
+                {
+                    int v;
+
+                    try
+                    {
+                        v = int.Parse(retryValues[i], CultureInfo.InvariantCulture);
+                    }
+                    catch (System.FormatException)
+                    {
+                        v = 0;
```

Comment:
I know you just moved this code but this seems like an odd thing to do. Just silently set to 0 if we can't parse it.

Reply:
It's the same logic in `Properties.getPropertyAsIntWithDefault` that you recently updated.